api/src/pcapi/scripts/transform_virtual_venues/main.py

```diff
@@ -0,0 +1,248 @@
+
+    return siret_info
+
+
+def get_address(venue: Venue, siret_info: SiretInfo) -> Address:
+    """Search address based on siret_info's geographical data.
+
+    If nothing is found, use the BAN API and create the missing address.
+    """
+    addresses = search_addresses(
+        street=siret_info.address.street, city=siret_info.address.city, postal_code=siret_info.address.postal_code
+    )
+
+    if addresses:
+        return addresses[0]
+
+    ban_address = _get_ban_address_or_none(
+        street=siret_info.address.street, postal_code=siret_info.address.postal_code, city=siret_info.address.city
+    )
+
+    if ban_address:
+        location_data = LocationData(
+            street=ban_address.street,
+            city=ban_address.city,
+            postal_code=ban_address.postcode,
+            insee_code=ban_address.citycode,
+            latitude=ban_address.latitude,
+            longitude=ban_address.longitude,
+            ban_id=ban_address.id,
+        )
+    else:
+        msg = (
+            f"venue #{venue.id} / "
+            f"street: '#{siret_info.address.street}' / "
+            f"city: '#{siret_info.address.city}' / "
+            f"postal code: '#{siret_info.address.postal_code}'"
+        )
+        raise FetchAddressError(msg)
+
+    return get_or_create_address(
+        location_data=location_data,
+        is_manual_edition=ban_address is None,
+    )
+
+
+def transform_virtual_venue(venue: Venue) -> None:
+    if not venue.isVirtual:
+        raise NotAVirtualVenue(f"venue #{venue.id}")
+
+    siret_info = get_siret(venue)
+    address = get_address(venue, siret_info)
+    offerer_address = get_or_create_offerer_address(offerer_id=venue.managingOffererId, address_id=address.id)
+
+    venue.siret = siret_info.siret
+
+    venue.isVirtual = False
+    venue.venueTypeCode = VenueTypeCode.ADMINISTRATIVE
+
+    venue.street = address.street  # type: ignore[method-assign]
+    venue.departementCode = address.departmentCode
+    venue.postalCode = address.postalCode
+    venue.city = address.city
+    venue.latitude = address.latitude
+    venue.longitude = address.longitude
+    venue.offererAddress = offerer_address
+
+    db.session.flush()
+
+
+def transform_venues_from_offerers_with_one_unique_virtual_venue() -> Collection[int]:
+    def transform_commit_and_log(venue: Venue) -> int | None:
+        logger.info("start venue transformation... %s", venue.id)
+
+        try:
+            with transaction():
+                transform_virtual_venue(venue)
+        except Exception as err:  # pylint: disable=broad-exception-caught
+            logger.info("venue transformation for %s failed because of: %s", venue.id, err)
+            return None
+
+        logger.info("venue transformation for %s done!", venue.id)
+        return venue.id
+
+    venues = venues_from_offerers_with_one_unique_virtual_venue()
+    ids = [transform_commit_and_log(venue) for venue in venues]
+    return {_id for _id in ids if _id}
```

Comment:
Peut-être tout simplement (vu que l'on n'utilise pas les `ids` ; à ce moment là il faut changer la signature de la fonction `-> None`):
 
```suggestion
    for venue in venues:
        transform_commit_and_log(venue)
```

Reply:
On s'en sert dans les tests, mais c'est vrai qu'autrement ce n'est pas utilisé.